src/pages/SyncProviderPage/SyncProviderPage.tsx

```diff
@@ -0,0 +1,60 @@
+import gt from 'lodash/gt'
+import { Redirect } from 'react-router-dom'
+
+import { useInternalUser } from 'services/user/useInternalUser'
+import { useFlags } from 'shared/featureFlags'
+import { loginProviderToShortName } from 'shared/utils/loginProviders'
+
+import SyncButton from './SyncButton'
+
+function SyncProviderPage() {
+  const { sentryLoginProvider } = useFlags({
+    sentryLoginProvider: false,
+  })
+
+  const { data: internalUser } = useInternalUser({})
+
+  // will be false if 0 | undefined | null
+  // will be true if greater than 1
+  const hasSynced = gt(internalUser?.owners?.length, 0)
+
+  // block all requests if flag is false
+  // --
+  // this allows us to also show the sync page within the limited
+  // experience env because if they don't have any providers selected
+  // then we need to have them select one before they move onto onboarding
+  // --
+  // once we allow users to sync multiple providers need change this logic
+  // to allow them to view this page
+  if (!sentryLoginProvider || hasSynced) {
```

Comment:
so in `useUserAcessGate` -> line 88:

>  if (sentryLoginProvider && !isGuest && !onSyncPage) {
>     // owners array contains a list of the synced providers
>     // if it is zero then they haven't synced any other providers
>     redirectToSyncPage = isEqual(internalUser?.owners?.length, 0)
>   }

you're already checking for this logic there

Reply:
so doesn't this mean the page will only be rendered if a user hasn't synced any providers? `useUserAccessGate` takes care of the redirect to /sync, will there be ever a case where the sync page is rendered when a user has already synced a provider?

Reply:
Yes in the future, we are planning to allow users to sync multiple providers